### PR TITLE
Fusetools 2684 environment page upgrade

### PR DESCRIPTION
**Pull request by apupier**
# Pull Request Checklist

After this checklist is all checked or PR provides explanations for possible pass-through, please put the label "Ready for review" on the PR.

## General

- [x] Did you use the Jira Issue number in the commit comments?
- [x] Did you set meaningful commit comments on each commit?
- [x] Did you sign off all commits for this PR? (git commit -s -m "jira issue number - your commit comment")

## Functional

- [x] Did the CI job report a successful build?
- [x] Is the non-happy path working, too?

## Maintainability

- [x] Are all Sonar reported issues fixed or can they be ignored?
- [x] Is there no duplicated code?
- [x] Are method-/class-/variable-names meaningful?

## Tests

- [x] Are there unit-tests?
- [x] Are there integration tests (or at least a jira to tackle these)?
- [ ] Do we need a new UI test?

## Legal

- [x] Have you used the correct file header copyright comment?
- [x] Have you used the correct year in the headers of new files?

### CODE REVIEW

**Comment by apupier** [comment]
created to have tests running in parallel of my local machine

**Comment by apupier** [comment]
image
image

image
image
image

**Comment by apupier** [comment]
image

**Comment by cunningt** [comment]
testPR

**Comment by apupier** [comment]
depends on https://github.com/jbosstools/jbosstools-fuse/pull/1298

File: editor/plugins/org.fusesource.ide.projecttemplates/src/org/fusesource/ide/projecttemplates/adopters/AbstractProjectTemplate.java [hunk 1]
```diff
@@ -91,7 +92,7 @@ private void refreshProjectSync(IProject project, IProgressMonitor monitor) thro
 	 */
 	public abstract TemplateCreatorSupport getCreator(CommonNewProjectMetaData projectMetaData);
 
-	public boolean isCompatible(String camelVersion) {
+	public boolean isCompatible(EnvironmentData environment) {
```
[the remark this answers is not in the document]
Comment (apupier) [reply editor/plugins/org.fusesource.ide.projecttemplates/src/org/fusesource/ide/projecttemplates/adopters/AbstractProjectTemplate.java]: the parameter is used in subclasses

File: editor/plugins/org.fusesource.ide.projecttemplates/src/org/fusesource/ide/projecttemplates/wizards/pages/model/TemplateItem.java [hunk 2]
```diff
@@ -38,13 +40,14 @@
 	 * @param category		the category
 	 * @param template		the template class
 	 */
-	public TemplateItem(String id, String name, String description, int weight, CategoryItem category, AbstractProjectTemplate template, String keywords) {
+	public TemplateItem(String id, String name, String description, int weight, CategoryItem category, AbstractProjectTemplate template, String keywords, CamelDSLType dslType) {
```
[the remark this answers is not in the document]
Comment (apupier) [reply editor/plugins/org.fusesource.ide.projecttemplates/src/org/fusesource/ide/projecttemplates/wizards/pages/model/TemplateItem.java]: argh.
is it worthy to create a Builder or a class of a subset of the parameters?
I'm not sure.

File: editor/plugins/org.fusesource.ide.projecttemplates/src/org/fusesource/ide/projecttemplates/wizards/pages/model/TemplateItem.java [hunk 3]
```diff
@@ -10,80 +10,54 @@
...
+		this.dslType = dslType;
 	}
 	
 	/**
 	 * @return the keywords
 	 */
 	public List<String> getKeywords() {
-		return this.keywords;
+		return templateItemIdentity.getKeywords();
 	}
```
[the remark this answers is not in the document]
Comment (apupier) [reply editor/plugins/org.fusesource.ide.projecttemplates/src/org/fusesource/ide/projecttemplates/wizards/pages/model/TemplateItem.java]: I refactored it, this comment is no more accurate

**Comment by bfitzpat** [comment]
A question... Standalone + Spring Boot is a valid combo?

**Comment by apupier** [comment]
> A question... Standalone + Spring Boot is a valid combo?

yes it is a valid one.
Management decided to use the same template for SpringBoot  standalone and SpringBoot on OpenShift.
The template will be improved (and the description of it) with this task: https://issues.jboss.org/browse/FUSETOOLS-2657

**Comment by apupier** [comment]
testPR

**Comment by bfitzpat** [comment]
+1 with green build :)

**Comment by apupier** [comment]
I set the label to "Ready for merge":
- Brian approved changes
- all other checks passed
--> waiting next master build to be green before merge

**Pull request merged**